### PR TITLE
Minor improvements to CONTRIBUTING.md; no substantial content changes

### PR DESCRIPTION
**Pull request by lancestarr**
As I was going through `CONTRIBUTING.md`, I made small corrections and enforced some consistency with headers. None of these changes are technical changes; they're all proofreading-type modifications.

It looks like the "found a bug" link (anchor) was dead on arrival, so that remains nonfunctional. Any suggestions on fixing that? I'd be happy to incorporate them, or you can make the fix yourself.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
 
-3. Follow the [Contribution Guidelines](#contribution-guidelines) to start working on the issue.
+3. Follow the [Contribution guidelines](#contribution-guidelines) to start working on the issue
 
 Remember to feel free to ask for help by leaving a comment within the Issue.
 
-Working on your first Pull Request? You can learn how from this *free* series [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
+Working on your first pull request? You can learn how from this *free* series [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github).
 
 ###### If you've found a bug that is not on the board, [follow these steps](#found-a-bug).
```
Comment (lancestarr) [inline CONTRIBUTING.md]: This link is (and was already) broken.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
+- [Running the test suite](#running-the-test-suite)
+- [Squashing your commits](#squashing-your-commits)
+- [Creating a pull request](#creating-a-pull-request)
+- [Common steps](#common-steps)
+- [How we review and merge pull requests](#how-we-review-and-merge-pull-requests)
+- [How we close stale issues](#how-we-close-stale-issues)
+- [Next steps](#next-steps)
 - [Other resources](#other-resources)
 - [Advice](#advice)
 - [Developing in an IDE](#developing-in-an-ide)
```
Comment (lancestarr) [inline CONTRIBUTING.md]: At the end of this table of contents, do we want to add links to "Publishing a new release" and "Acknowledgment"?

File: CONTRIBUTING.md (hunk 1 above)
Comment (tleonhardt) [reply CONTRIBUTING.md]: Good catch!  My intent was to add a "Found a Bug?" section towards the end of this contributing guide and it looks like I forgot to do that.

I was intending it to look something like the following:

### Found a bug?

If you think you've found a bug, please first read through the open [Issues](https://github.com/python-cmd2/cmd2/issues). If you're confident it's a new bug, go ahead and create a new GitHub issue. Be sure to include as much information as possible so we can reproduce the bug.  At a minimum, please state the following:

* ``cmd2`` version
* Python version
* OS name and version

File: CONTRIBUTING.md (hunk 2 above)
Comment (tleonhardt) [reply CONTRIBUTING.md]: Yes please.  And also the "Found a bug" section.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
-- [Make Changes](#make-changes)
-- [Static Code Analysis](#static-code-analysis)
-- [Run The Test Suite](#run-the-test-suite)
-- [Squash Your Commits](#squash-your-commits)
-- [Creating A Pull Request](#creating-a-pull-request)
-- [Common Steps](#common-steps)
-- [How We Review and Merge Pull Requests](#how-we-review-and-merge-pull-requests)
-- [How We Close Stale Issues](#how-we-close-stale-issues)
-- [Next Steps](#next-steps)
+- [Forking the project](#forking-the-project)
```
Comment (tleonhardt) [inline CONTRIBUTING.md]: You make me realize how much I unnecessarily capitalize things.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tleonhardt) [reply CONTRIBUTING.md]: I added the link to the "Publishing a new release" section, but opted to put the "Found a bug" section at the end of the Readme.md file instead.

**Pull request merged**